test/handlers/fee/dynamic/collectFeeGenericEVM.js

```diff
@@ -0,0 +1,401 @@
+            ber*10^18:      uint256
+            ter*10^18:      uint256
+            dstGasPrice:    uint256
+            expiresAt:      uint256
+            fromDomainID:   uint8 encoded as uint256
+            toDomainID:     uint8 encoded as uint256
+            resourceID:     bytes32
+            msgGasLimit:    uint256
+            sig:            bytes(65 bytes)
+
+        total in bytes:
+        message:
+            32 * 8  = 256
+        message + sig:
+            256 + 65 = 321
+
+            amount: uint256
+        total feeData length: 353
+    */
+
+  beforeEach(async () => {
+    await Promise.all([
+      (BridgeInstance = await Helpers.deployBridge(
+        originDomainID,
+        accounts[0]
+      ))
+    ]);
+
+    PermissionlessGenericHandlerInstance = await PermissionlessGenericHandlerContract.new(
+      BridgeInstance.address
+    );
+    FeeHandlerRouterInstance = await FeeHandlerRouterContract.new(
+      BridgeInstance.address
+    );
+    DynamicGenericFeeHandlerEVMInstance = await DynamicGenericFeeHandlerEVMContract.new(
+      BridgeInstance.address,
+      FeeHandlerRouterInstance.address
+    );
+    TestStoreInstance = await TestStoreContract.new();
+
+    resourceID = Helpers.createResourceID(
+      TestStoreInstance.address,
+      originDomainID
+    );
+
+    depositFunctionSignature = Helpers.getFunctionSignature(
+      TestStoreInstance,
+      "storeWithDepositor"
+    );
+
+    await Promise.all([
+      BridgeInstance.adminSetResource(
+        PermissionlessGenericHandlerInstance.address,
+        resourceID,
+        TestStoreInstance.address,
+        emptySetResourceData
+      ),
+      BridgeInstance.adminChangeFeeHandler(FeeHandlerRouterInstance.address),
+      FeeHandlerRouterInstance.adminSetResourceHandler(
+        destinationDomainID,
+        resourceID,
+        DynamicGenericFeeHandlerEVMInstance.address
+      ),
+    ]);
+
+    depositData = Helpers.createPermissionlessGenericDepositData(
+      depositFunctionSignature,
+      TestStoreInstance.address,
+      destinationMaxFee,
+      depositorAddress,
+      hashOfTestStore
+    );
+
+    await DynamicGenericFeeHandlerEVMInstance.setFeeOracle(oracle.address);
+
+    // set MPC address to unpause the Bridge
+    await BridgeInstance.endKeygen(Helpers.mpcAddress);
+  });
+
+  it("should collect fee in native tokens", async () => {
+    const oracleResponse = {
+      ber: Ethers.utils.parseEther("0.000533"),
+      ter: Ethers.utils.parseEther("1.63934"),
+      dstGasPrice: Ethers.utils.parseUnits("30000000000", "wei"),
+      expiresAt: Math.floor(new Date().valueOf() / 1000) + 500,
+      fromDomainID: originDomainID,
+      toDomainID: destinationDomainID,
+      resourceID,
+      msgGasLimit,
+    };
+
+    const feeData = Helpers.createOracleFeeData(
+      oracleResponse,
+      oracle.privateKey,
+      tokenAmount
+    );
+
+    const {fee} = await FeeHandlerRouterInstance.calculateFee.call(
```

Comment:
If the fee is known to be "0.000036777" then perhaps no need to call calculateFee() ?